code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -359,6 +359,10 @@
 					if(isturf(S.loc) || ishuman(S.loc))
 						movement_target = S
 						break
+				for(var/obj/item/toy/rubber_ball/B in oview(src, 4))
```

Comment:
Working on this right now. Should I just be putting all the searching and "playing" code under Ian's subtype or make a list of objects in the simple_animal base mob that they would seek out and interact with?